{
  "comments": [
    {
      "unresolved": false,
      "key": {
        "uuid": "c0164e61_172c2387",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 756
      },
      "writtenOn": "2022-09-22T22:25:56Z",
      "side": 1,
      "message": "By default deployed instances need to be accessed with \u0027Host: dashboard-ci.tripleo.org\u0027 at CS8: 10.0.108.102 and CS9: 10.0.108.148",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "47f0d646_12089fe0",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 192
      },
      "writtenOn": "2022-09-23T06:08:56Z",
      "side": 1,
      "message": "big change it will need a few visits\noverall lgtm on first pass",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "e0128626_c1d3e06e",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 753
      },
      "writtenOn": "2022-09-23T13:44:08Z",
      "side": 1,
      "message": "As discussed yesterday [1], dasm will try to use docker-compose with podman [2] instead of podman-compose first. This might help us to get rid of the hacks that dasm had to do to get podman-compose working. \n\nUsing \u0027workflow -1\u0027 simply to mark it as work in progress.\n\n[1] https://meetings.opendev.org/irclogs/%23oooq/%23oooq.2022-09-22.log.html#t2022-09-22T19:18:55\n[2] https://www.redhat.com/sysadmin/podman-docker-compose",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "09b976b7_adea7dbc",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 756
      },
      "writtenOn": "2022-09-28T15:20:40Z",
      "side": 1,
      "message": "docker-compose is causing a lot of errors:\n\n```\n[centos@rrcockpit files]$ GRAFANA_SMTP_PASSWORD\u003dtest GRAFANA_ADMIN_PASSWORD\u003dtest docker-compose up\nCreating network \"files_default\" with driver \"bridge\"\nERROR: Volume influxdb-volume declared as external, but could not be found. Please create the volume manually using `docker volume create --name\u003dinfluxdb-volume` and try again.\n[centos@rrcockpit files]$ docker volume create --name\u003dinfluxdb-volume\nError: unknown flag: --name\nSee \u0027podman volume create --help\u0027\n[centos@rrcockpit files]$ docker volume create influxdb-volume\nError: volume with name influxdb-volume already exists: volume already exists\n[centos@rrcockpit files]$ GRAFANA_SMTP_PASSWORD\u003dtest GRAFANA_ADMIN_PASSWORD\u003dtest docker-compose up\nERROR: Network \"files_default\" needs to be recreated - option \"isolate\" has changed\n```\n\nI spent last few days trying to mitigate all errors to no avail.",
      "parentUuid": "e0128626_c1d3e06e",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "c66b8c36_6d5165ef",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 753
      },
      "writtenOn": "2022-09-28T19:13:36Z",
      "side": 1,
      "message": "Dariusz asked me to remove the \u0027workflow -1\u0027 [1].\n\n[1] https://meetings.opendev.org/irclogs/%23oooq/%23oooq.2022-09-28.log.html#t2022-09-28T15:21:47\n\nThanks for testing docker-compose with podman, Dariusz!\n\nUsing podman-compose seems to have its own issues. It requires a lot of hacks to get it work and has drawbacks such as getting installed on every ansible-pull run, i.e. every 5 minutes.\n\nThe errors of docker-compose with podman cited below do not seem to be related to the combination of docker-compose and podman. But nevertheless it caused some friction for Dariusz, so we might want to continue using docker instead of podman which is well supported on centos (and seems to be part of rhel) and can easily be installed from Docker\u0027s yum repo [2].\n\n[2] https://docs.docker.com/engine/install/centos/\n",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "a3f423ee_a6868e02",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 753
      },
      "writtenOn": "2022-09-28T19:13:36Z",
      "side": 1,
      "message": "Hi Dariusz,\nthanks for testing docker-compose with podman! The guides did not hint at those issuesðŸ˜ž\n\n\u003e [centos@rrcockpit files]$ GRAFANA_SMTP_PASSWORD\u003dtest GRAFANA_ADMIN_PASSWORD\u003dtest docker-compose up\n\u003e Creating network \"files_default\" with driver \"bridge\"\n\u003e ERROR: Volume influxdb-volume declared as external, but could not be found. Please create the volume manually using `docker volume create --name\u003dinfluxdb-volume` and try again.\n\nUnrelated to docker-compose/podman-compose: Docker volumes are created in a separate step in role rrcockpit [1]. *-compose fails because those have not been created on your testsystem. Or maybe you did not run these commands as root?\n\n[1] https://github.com/rdo-infra/ci-config/blob/76c2bec8c58f9fad480462bfb1fabc8a51278f5c/ci-scripts/infra-setup/roles/rrcockpit/tasks/main.yml#L6\n\n\u003e [centos@rrcockpit files]$ docker volume create --name\u003dinfluxdb-volume\n\u003e Error: unknown flag: --name\n\u003e See \u0027podman volume create --help\u0027\n\nThis is the wrong syntax for both docker [1] and podman-docker [2], use e.g. \u0027docker volume create influxdb-volume\u0027.\n\n[1] https://manpages.debian.org/bullseye/docker.io/docker-volume-create.1.en.html\n[2] https://manpages.debian.org/bullseye/podman/podman-volume-create.1.en.html\n\n\u003e [centos@rrcockpit files]$ docker volume create influxdb-volume\n\u003e Error: volume with name influxdb-volume already exists: volume already exists\n\nSomething is broken on your system. Did you really executed the previous commands in a row without other commands in between?\n\n\u003e [centos@rrcockpit files]$ GRAFANA_SMTP_PASSWORD\u003dtest GRAFANA_ADMIN_PASSWORD\u003dtest docker-compose up\n\u003e ERROR: Network \"files_default\" needs to be recreated - option \"isolate\" has changed\n\u003e ```\n\nLeftovers from previous runs?! Better start with a fresh environment:\n\n docker-compose down --remove-orphans \u0026\u0026 docker-compose up -d",
      "parentUuid": "09b976b7_adea7dbc",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "7b565b9d_26bbe0ed",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 756
      },
      "writtenOn": "2022-09-28T20:13:29Z",
      "side": 1,
      "message": "ad 1.\ndocker volumes are created before docker-compose starts.\nad 2.\nAs you commented below, I used both syntaxes. First, hinted by docker-compose and second, actual syntax. However, both are causing issues.\nad 3.\ninfluxdb-volume is marked as external volume, hence is not removed via docker-compose. It needs to be created by ansible. After creating that, docker-compose complains about it being inaccessible.",
      "parentUuid": "a3f423ee_a6868e02",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "babea20f_d5fa50f5",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 749
      },
      "writtenOn": "2022-09-28T20:32:06Z",
      "side": 1,
      "message": "\u003e ERROR: Volume influxdb-volume declared as external, but could not be found. Please create the volume manually using `docker volume create --name\u003dinfluxdb-volume` and try again.\n\nDid you make sure you created the volumes as root? I don\u0027t think docker-compose would use rootless containers.",
      "parentUuid": "7b565b9d_26bbe0ed",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "333f9f86_c57f20d9",
        "filename": "ci-scripts/infra-setup/grafana_playbook.yml",
        "patchSetId": 4
      },
      "lineNbr": 31,
      "author": {
        "id": 192
      },
      "writtenOn": "2022-09-23T06:08:56Z",
      "side": 1,
      "message": "if this is really just for Centos8 (per task name) then make it \u003d\u003d",
      "range": {
        "startLine": 31,
        "startChar": 62,
        "endLine": 31,
        "endChar": 64
      },
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "7023928a_b0552d6a",
        "filename": "ci-scripts/infra-setup/grafana_playbook.yml",
        "patchSetId": 4
      },
      "lineNbr": 71,
      "author": {
        "id": 192
      },
      "writtenOn": "2022-09-23T06:08:56Z",
      "side": 1,
      "message": "path changed? i assume this is the replacement for \u0027create files\u0027 task you are removing just above but you are moving them to /tmp where before they were going to /root",
      "range": {
        "startLine": 71,
        "startChar": 14,
        "endLine": 71,
        "endChar": 15
      },
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "b230efdb_003c5b72",
        "filename": "ci-scripts/infra-setup/grafana_playbook.yml",
        "patchSetId": 4
      },
      "lineNbr": 71,
      "author": {
        "id": 756
      },
      "writtenOn": "2022-09-28T15:20:40Z",
      "side": 1,
      "message": "\"Create files\" war creating and copying files with passwords. They were later read and used as a part of containers bring up.\nThis \"copy files\" is the way to provide local files from \"infra-setup\" to run on remote host (rrcockpit).\nI haven\u0027t found the way to remotely run docker-compose without copying files.",
      "parentUuid": "7023928a_b0552d6a",
      "range": {
        "startLine": 71,
        "startChar": 14,
        "endLine": 71,
        "endChar": 15
      },
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "b313587d_1b3d4b9e",
        "filename": "ci-scripts/infra-setup/roles/podman_compose/tasks/main.yaml",
        "patchSetId": 4
      },
      "lineNbr": 36,
      "author": {
        "id": 753
      },
      "writtenOn": "2022-09-28T19:13:36Z",
      "side": 1,
      "message": "This will download and install podman-compose from github every 5 minutes?!\n\nTest locally with\n\n pip install https://github.com/containers/podman-compose/archive/devel.tar.gz -v -v -v",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "54a49554_b2d952d4",
        "filename": "ci-scripts/infra-setup/roles/podman_compose/tasks/main.yaml",
        "patchSetId": 4
      },
      "lineNbr": 43,
      "author": {
        "id": 753
      },
      "writtenOn": "2022-09-28T19:13:36Z",
      "side": 1,
      "message": "Looks like this role podman_compose, role rrcockpit and its tasks file start_services.yml are still run as root. Podman will be run as root so containers can expose port 80. So why allowing access to port 80 for unprivileged users?",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "bcc62968_5d940949",
        "filename": "ci-scripts/infra-setup/roles/podman_compose/tasks/main.yaml",
        "patchSetId": 4
      },
      "lineNbr": 43,
      "author": {
        "id": 756
      },
      "writtenOn": "2022-09-28T20:10:04Z",
      "side": 1,
      "message": "podman runs rootless, hence nginx cannot access port 80.\nThis change is going to expose ports 80+ to all unprivileged users. What kind of risk are you thinking about with this type of exposure?",
      "parentUuid": "54a49554_b2d952d4",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "ea5432fc_f7f23a0d",
        "filename": "ci-scripts/infra-setup/roles/podman_compose/tasks/main.yaml",
        "patchSetId": 4
      },
      "lineNbr": 59,
      "author": {
        "id": 753
      },
      "writtenOn": "2022-09-28T19:13:36Z",
      "side": 1,
      "message": "But this patch does not run containers as rootless, do they?",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "26eba429_028f31f0",
        "filename": "ci-scripts/infra-setup/roles/podman_compose/tasks/main.yaml",
        "patchSetId": 4
      },
      "lineNbr": 59,
      "author": {
        "id": 756
      },
      "writtenOn": "2022-09-28T20:10:04Z",
      "side": 1,
      "message": "podman is configured to run rootless. Only centos or cloud-user will have access.",
      "parentUuid": "ea5432fc_f7f23a0d",
      "revId": "6866e979744fbe4c7f852d405e9e86ec9411c42f",
      "serverId": "270e2033-b340-4cff-9539-693957ebf0e7"
    }
  ]
}